src/herbivoresAndCarnivores.js

```diff
@@ -1,16 +1,45 @@
 'use strict';
 
 class Animal {
-  // write your code here
-}
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+  };
+};
+
+Animal.alive = [];
 
 class Herbivore extends Animal {
-  // write your code here
-}
+  constructor(name, health, hidden) {
+    super(name);
+
+    if (hidden === undefined) {
+      this.hidden = false;
+    };
+
+    Animal.alive.push(this);
+  };
+
+  hide() {
+    this.hidden = true;
+  };
+};
 
 class Carnivore extends Animal {
-  // write your code here
-}
+  constructor(name) {
+    super(name);
+
+    Animal.alive.push(this);
+  };
+
+  bite(herb) {
+    if (herb.hidden === false) {
```

Comment:
```suggestion
  bite(animal) {
    if (!animal.hidden) {
```

Reply:
also is should be Herbivore, not every animal can be bitten according to the task